data_steward/utils/auth.py

```diff
@@ -0,0 +1,70 @@
+# Python imports
+import requests as req
+
+# Third party imports
+from google.auth import iam
+from google.auth.transport import requests
+from google.oauth2 import service_account
+
+METADATA_URL = 'http://metadata.google.internal/computeMetadata/v1/'
+METADATA_HEADERS = {'Metadata-Flavor': 'Google'}
+SERVICE_ACCOUNT = 'default'
+TOKEN_URI = 'https://accounts.google.com/o/oauth2/token'
+
+
+def get_access_token(scopes):
+    """
+    Retrieves an access_token in App Engine for the default service account
+
+    Source: https://cloud.google.com/functions/docs/securing/function-identity#gcloud
+    and https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances#applications
+    :param scopes: List of Google scopes as strings
+    :return: access token as string to be used in a request header as
+        headers = {'Authorization': f'Bearer {access_token}'}
+    """
+    scopes_str = ','.join(scopes)
+    url = f'{METADATA_URL}instance/service-accounts/{SERVICE_ACCOUNT}/token?scopes={scopes_str}'
+    # Request an access token from the metadata server.
+    r = req.get(url, headers=METADATA_HEADERS)
+    r.raise_for_status()
+    # Extract the access token from the response.
+    access_token = r.json()['access_token']
+    return access_token
+
+
+def delegated_credentials(credentials, subject, scopes):
```

Comment:
nit: subject can be an optional param
```suggestion
def delegated_credentials(credentials, scopes, subject=None):
```